### ISSUE TITLE
[docs] Wrong return type for srt_setsockopt() in the API.md

### ISSUE THREAD

**Issue by nh2** [Haivision/srt#537]
I notice many options to `srt_setsockopt()` have constraints on them, for example for the `passphrase`:

https://github.com/Haivision/srt/blame/6404396588fbc81b6c0f998463da7df479ce31a4/docs/API.md#L599

> Range `[10..79]`

It seems ([issue report in gstreamer](https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/issues/694#note_106624)) that if the passphrase is too short, it is silently ignored, and the stream is sent unencrypted!

Is there no way for the `srt_setsockopt()` function to communicate errors to the caller, or did I miss it somehow?

It seems sensible to have that, given that many of the options have validity constraints on them.

**Comment by nh2** [comment]
Ah, I think your [docs](https://github.com/Haivision/srt/blame/6404396588fbc81b6c0f998463da7df479ce31a4/docs/API.md#L363) are wrong!

In there you say:

```
    void srt_getsockopt(SRTSOCKET socket, int level, SRT_SOCKOPT optName, void* optval, int& optlen);
    void srt_setsockopt(SRTSOCKET socket, int level, SRT_SOCKOPT optName, const void* optval, int optlen);
```

but in `srt.h` actually the function returns `int`, not `void`, and the `int` is for the error code.

* [`srt_setsocketopt()` code](https://github.com/Haivision/srt/blob/6404396588fbc81b6c0f998463da7df479ce31a4/srtcore/srt_c_api.cpp#L103-L104)
* [`CUDT::setsockopt()` implementation that shows `return ERROR`](https://github.com/Haivision/srt/blob/6404396588fbc81b6c0f998463da7df479ce31a4/srtcore/api.cpp#L2138-L2158)

I think that as a result of the docs being wrong, the gstreamer bindings are completely wrong, because they ignore the returned error codes everywhere.

**Comment by ethouris** [comment]
Thanks for finding it. Yes, in case of error in the option, both these functions return -1 and set an error available at `srt_getlasterror()`. Unfortunately there's no other way of preventing the stream from running unencrypted than just check for errors reported from these functions.

**Comment by maxlovic** [comment]
Fixed by 239f8c92ce0f9cf24e95be8da23424cd618e533a

**Comment by nh2** [comment]
Thanks!
